breadth_first_search/BreadthFirstSearch.java

```diff
@@ -0,0 +1,65 @@
+import java.util.*;
+
+public class BreadthFirstSearch <T> {
+
+    // HasMap  of lists for Adjacency List Representation
+    public HashMap<Integer, ArrayList<T>> adj;
```

Comment:
>just modifying a single line 

means?

> This method was always a bad idea.
?
>You'll have to tackle the situation that user may be adding an edge between two unregistered nodes.

It can be add if you just give input in initEdges.
or
Do you mean by to remove initEdges?